src/components/Dashboard/Dashboard.js

```diff
@@ -42,101 +51,96 @@ const dashboardMap = {
+        return (
+            <Layer translucent>
+                <CenteredContent>
+                    <CircularLoader />
+                </CenteredContent>
+            </Layer>
+        )
     }
 
-    setHeaderbarVisibility = () => {
-        const header = document.getElementsByTagName('header')[0]
-        if (isPrintMode(this.props.mode)) {
-            header.classList.add('hidden')
-        } else {
-            header.classList.remove('hidden')
-        }
+    if (mode === NEW) {
+        return dashboardMap[mode]
     }
 
-    componentDidMount() {
-        window.onresize = debounce(
-            () => this.props.setWindowHeight(window.innerHeight),
-            300
+    if (dashboardsIsEmpty) {
+        return (
+            <>
+                <DashboardsBar />
+                <DashboardVerticalOffset />
+                <NoContentMessage
+                    text={i18n.t(
+                        'No dashboards found. Use the + button to create a new dashboard.'
+                    )}
+                />
+            </>
         )
-        this.setDashboard()
     }
 
-    componentDidUpdate() {
-        this.setDashboard()
+    if (id === NON_EXISTING_DASHBOARD_ID) {
+        return (
+            <>
+                <DashboardsBar />
+                <DashboardVerticalOffset />
+                <NoContentMessage
+                    text={i18n.t('Requested dashboard not found')}
+                />
+            </>
+        )
     }
 
-    render() {
-        const { id, mode, dashboardsLoaded, dashboardsIsEmpty } = this.props
-
-        if (!dashboardsLoaded || id === null) {
-            return (
-                <Layer translucent>
-                    <CenteredContent>
-                        <CircularLoader />
-                    </CenteredContent>
-                </Layer>
-            )
-        }
-
-        if (mode === NEW) {
-            return dashboardMap[mode]
-        }
-
-        if (dashboardsIsEmpty) {
-            return (
-                <>
-                    <DashboardsBar />
-                    <DashboardVerticalOffset />
-                    <NoContentMessage
-                        text={i18n.t(
-                            'No dashboards found. Use the + button to create a new dashboard.'
-                        )}
-                    />
-                </>
-            )
-        }
-
-        if (id === NON_EXISTING_DASHBOARD_ID) {
-            return (
-                <>
-                    <DashboardsBar />
-                    <DashboardVerticalOffset />
-                    <NoContentMessage
-                        text={i18n.t('Requested dashboard not found')}
-                    />
-                </>
-            )
-        }
-
-        return dashboardMap[mode]
-    }
+    return dashboardMap[mode]
 }
 
 Dashboard.propTypes = {
     dashboardsIsEmpty: PropTypes.bool,
     dashboardsLoaded: PropTypes.bool,
     id: PropTypes.string,
-    match: PropTypes.object,
```

Comment:
Hmm... isn't this still needed, technically, for the final exported component?  As is there's no indication that a `match` object is required, and

Reply:
Restored